osu.Game/Screens/Play/KeyCounter.cs

```diff
@@ -69,82 +67,36 @@ public void Decrement()
-        public Color4 KeyDownTextColor { get; set; } = Color4.DarkGray;
-        public Color4 KeyUpTextColor { get; set; } = Color4.White;
-        public double FadeTime { get; set; }
-
-        protected KeyCounter(string name)
+        public abstract partial class InputTrigger : Component
         {
-            Name = name;
-        }
+            private KeyCounter? target;
 
-        [BackgroundDependencyLoader(true)]
-        private void load(TextureStore textures)
-        {
-            Children = new Drawable[]
+            public KeyCounter Target
             {
-                buttonSprite = new Sprite
-                {
-                    Texture = textures.Get(@"KeyCounter/key-up"),
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                },
-                glowSprite = new Sprite
-                {
-                    Texture = textures.Get(@"KeyCounter/key-glow"),
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    Alpha = 0
-                },
-                textLayer = new Container
-                {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    RelativeSizeAxes = Axes.Both,
-                    Children = new Drawable[]
-                    {
-                        new OsuSpriteText
-                        {
-                            Text = Name,
-                            Font = OsuFont.Numeric.With(size: 12),
-                            Anchor = Anchor.Centre,
-                            Origin = Anchor.Centre,
-                            RelativePositionAxes = Axes.Both,
-                            Position = new Vector2(0, -0.25f),
-                            Colour = KeyUpTextColor
-                        },
-                        countSpriteText = new OsuSpriteText
-                        {
-                            Text = CountPresses.ToString(@"#,0"),
-                            Anchor = Anchor.Centre,
-                            Origin = Anchor.Centre,
-                            RelativePositionAxes = Axes.Both,
-                            Position = new Vector2(0, 0.25f),
-                            Colour = KeyUpTextColor
-                        }
-                    }
-                }
-            };
-            // Set this manually because an element with Alpha=0 won't take it size to AutoSizeContainer,
-            // so the size can be changing between buttonSprite and glowSprite.
-            Height = buttonSprite.DrawHeight;
-            Width = buttonSprite.DrawWidth;
-        }
+                set => target = value;
+            }
 
-        private void updateGlowSprite(bool show)
-        {
-            if (show)
+            protected InputTrigger(string name)
+            {
+                Name = name;
+            }
+
+            protected void Light(bool increment = true)
```

Comment:
My favourites, bool arguments, and opaque ones at that. At face value nobody is going to know as to whether they want to "increment" or "preserve" or whatever. As far as I gather this is essentially for rewind? Then I'd call it by name:

```csharp
protected void Activate(bool forwardPlayback = true);
```

or even better yet

```csharp
protected void Activate(int playbackDirection);
```

and then you do `Activate(1)` when going forwards and `Activate(-1)` when doing backwards.

Reply:
The worst part is the `Unlight` method which has a different named argument which is doing the ... same thing in reverse? My brain hurt.

Reply:
@ItsShamed This is not fully addressed; while `InputTrigger` looks better wrt naming, `Activate(bool increment)` and `Deactivate(bool preserve)` are still there.

Reply:
Oh yeah, my brain trolled me :trollface: 
I thought I did a full pass in c61fac578ca53fa4dba22ac7ec85aa0cc335c762